src/main/java/lotto/domain/lotto/LottoNumberGenerator.java

```diff
@@ -0,0 +1,22 @@
+package lotto.domain.lotto;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class LottoNumberGenerator {
+    private static final List<Integer> numbers = IntStream.range(Lotto.MIN_NUMBER, Lotto.MAX_NUMBER + 1)
+            .boxed()
+            .collect(Collectors.toList());
+
+    private LottoNumberGenerator() {
+    }
+
+    public static List<Integer> getNumbers() {
+        Collections.shuffle(numbers);
+        return numbers.stream()
+                .limit(Lotto.NUMBER_COUNT)
+                .collect(Collectors.toList());
```

Comment:
Collections.unmodifiablelist()를 활용하여 Collection을 read-only로 리턴하여 외부에서 값의 변조를 막을 수 있도록 해주시면 좋을 것 같아요!

```suggestion
.collect(Collectors.collectingAndThen(Collectors.toList(), Collections::unmodifiableList));
```